### ISSUE TITLE
wrong pass phrase gets accepted on key_mismatch

### ISSUE THREAD

**Issue by tomholub** [FlowCrypt/flowcrypt-ios#523]
similar problem to #519 and #522 and another way to demonstrate #521

to reproduce:
1) set up app with pass phrase in session
2) restart app
3) open a message encrypted for another account (eg encrypt it for some other person then manually move the message to this account)

actual result: user is prompted for a pass phrase, and it accepts a wrong one, then shows `key_mismatch` error

expected result: as per #521 the wrong pass phrase should get rejected. Then when I enter the right pass phrase, I should see `key_mismatch` error.

**Comment by tomholub** [comment]
I suppose #519 as re-written should close this issue, but I'll leave this one open for further testing afterwards.

**Comment by tomholub** [comment]
assigning this to triage - open a message that was not encrypted for you, and see how pass phrases are handled, especially when giving a wrong one. Unless it no longer even asks? Unsure - please try and document it here.

**Comment by sosnovsky** [comment]
After #519 fix passhrase isn't asked anymore for mails with `key_mismatch` error.

I also added handling for `key_mismatch` errors by showing alert which allows to open undecrypted mail or just go back.
@tomholub should we allow users to see undecrypted mail or "go back" action will be enough?

![Simulator Screen Recording - iPhone 12 - 2021-10-18 at 13 04 40](https://user-images.githubusercontent.com/1062093/137710906-9c788e86-ad4a-4f48-8349-d1e597323c42.gif)

**Comment by tomholub** [comment]
You can merge it as you did it, but we will have to improve that later, in another issue.
